### ISSUE TITLE
System.Reflection.TargetInvocationException: Failed to load language 'IronPython 2.7.10'

### ISSUE THREAD

**Issue by DjArt** [IronLanguages/ironpython2#608]
### Prerequisites
* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

### Description

[Description of the bug or feature]
`System.Reflection.TargetInvocationException: Failed to load language 'IronPython 2.7.10': Operation is not supported on this platform. ---> System.PlatformNotSupportedException: Operation is not supported on this platform.    в System.Comparison`1.EndInvoke(IAsyncResult result) в E:\A\_work\222\s\corefx\src\System.Linq.Expressions\src\System\Linq\Expressions\Compiler\DelegateHelpers.cs:строка 130    в IronPython.Runtime.PythonContext.GetInitialPrefix()    в IronPython.Runtime.PythonContext..ctor(ScriptDomainManager manager, IDictionary`2 options)    в _$ILCT$.$ILT$ReflectionDynamicInvoke$.InvokeRetVII[T0,T1](Object thisPtr, IntPtr methodToCall, ArgSetupState& argSetupState, Boolean targetIsThisCall)    в System.InvokeUtils.CalliIntrinsics.Call(IntPtr dynamicInvokeHelperMethod, IntPtr dynamicInvokeHelperGenericDictionary, Object thisPtr, IntPtr methodToCall, ArgSetupState& argSetupState, Boolean isTargetThisCall)    в System.InvokeUtils.CallDynamicInvokeMethod(Object thisPtr, IntPtr methodToCall, Object thisPtrDynamicInvokeMethod, IntPtr dynamicInvokeHelperMethod, IntPtr dynamicInvokeHelperGenericDictionary, Object targetMethodOrDelegate, Object[] parameters, BinderBundle binderBundle, Boolean wrapInTargetInvocationException, Boolean invokeMethodHelperIsThisCall, Boolean methodToCallIsThisCall)    Exception_EndOfInnerExceptionStack    в Microsoft.Scripting.Runtime.LanguageConfiguration.LoadLanguageContext(ScriptDomainManager domainManager, Boolean& alreadyLoaded)    в Microsoft.Scripting.Runtime.DlrConfiguration.LoadLanguageContext(ScriptDomainManager manager, LanguageConfiguration config)    в Microsoft.Scripting.Runtime.DlrConfiguration.TryLoadLanguage(ScriptDomainManager manager, AssemblyQualifiedTypeName providerName, LanguageContext& language)    в Microsoft.Scripting.Runtime.ScriptDomainManager.GetLanguageByTypeName(String providerAssemblyQualifiedTypeName)    в Microsoft.Scripting.Hosting.ScriptRuntime.GetEngineByTypeName(String assemblyQualifiedTypeName)
`
### Steps to Reproduce

Compile and run simple UWP app in **Release mode**, that using IronPython.
That exception throws when `Assembly.CodeBase` property accessed. May be CoreRT affiliated too.
If `String.Empty` is permissible, I think exception should be catch.

**Comment by slozier** [comment]
UWP does not support Reflection.Emit (hence the PlatformNotSupportedException).

**Comment by DjArt** [comment]
Without compiling to native executable all works fine. I will make some tests.

**Comment by DjArt** [comment]
Ok, with updated DLR & some fixes simple oneline(aka `77 * 64`) scripts works fine.

**Comment by DjArt** [comment]
.NET Native compiler cuts some metadata for optimizations based on included code. So, for example, if `Func<,>` isn't used in any code in project(library or app), it can be just removed or be available for reflection(not instance creation) only. It can be fixed by [Runtime directives](https://docs.microsoft.com/en-us/dotnet/framework/net-native/runtime-directives-rd-xml-configuration-file-reference). So, we need list of assemblies, that are used by IronPython and they StdLib.

**Comment by slozier** [comment]
Can't use the default runtime directives?
```xml
<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
  <Application>
    <!--
      An Assembly element with Name="*Application*" applies to all assemblies in
      the application package. The asterisks are not wildcards.
    -->
    <Assembly Name="*Application*" Dynamic="Required All" />
    
    
    <!-- Add your application specific runtime directives here. -->


  </Application>
</Directives>
```

**Comment by DjArt** [comment]
No, because it's not included framework libraries. But, I think, that can handle most cases:
```
<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
    <Application>
        <Assembly Name="System" Dynamic="Required All"/>
        <Assembly Name="System.Core" Dynamic="Required All"/>
        <Assembly Name="System.Dynamic.Runtime" Dynamic="Required All"/>
        <Assembly Name="System.Collections" Dynamic="Required All"/>
        <Assembly Name="System.Collections.Immutable" Dynamic="Required All"/>
        <Assembly Name="System.Collections.NonGeneric" Dynamic="Required All"/>
        <Assembly Name="System.Collections.Specialized" Dynamic="Required All"/>
        <Assembly Name="System.Collections.Concurrent" Dynamic="Required All"/>
        <Assembly Name="System.Linq" Dynamic="Required All"/>
        <Assembly Name="System.Linq.Expressions" Dynamic="Required All"/>
        <Assembly Name="System.Reflection" Dynamic="Required All"/>
        <Assembly Name="System.Runtime" Dynamic="Required All"/>
        <Assembly Name="*Application*" Dynamic="Required All"/>
        <Type Name="System.Func`3[System.Runtime.CompilerServices.CallSite,System.Object,System.Collections.Generic.KeyValuePair`2[System.Collections.IEnumerator,System.IDisposable]]" Dynamic="Required All"/>
    </Application>
</Directives>
```
But, I think, it should be populated with some generics.

**Comment by slozier** [comment]
Hmm, there are likely thousands of different generic combinations used by IronPython...
